azure-iot-device/azure/iot/device/iothub/pipeline/mqtt_topic_iothub.py

```diff
@@ -46,22 +50,65 @@ def get_input_topic_for_subscribe(device_id, module_id):
     return _get_topic_base(device_id, module_id) + "/inputs/#"
 
 
+def get_method_topic_for_subscribe():
+    """
+    :return: The topic for ALL incoming methods. It is of the format
+    "$iothub/methods/POST/#"
+    """
+    return "$iothub/methods/POST/#"
+
+
+def get_twin_response_topic_for_subscribe():
+    """
+    :return: The topic for ALL incoming twin responses. It is of the format
+    "$iothub/twin/res/#"
+    """
+    return "$iothub/twin/res/#"
+
+
+def get_twin_patch_topic_for_subscribe():
+    """
+    :return: The topic for ALL incoming twin patches. It is of the format
+    "$iothub/twin/PATCH/properties/desired/#
+    """
+    return "$iothub/twin/PATCH/properties/desired/#"
+
+
+def get_telemetry_topic_for_publish(device_id, module_id):
+    """
+    return the topic string used to publish telemetry
+    """
+    return _get_topic_base(device_id, module_id) + "/messages/events/"
+
+
 def get_method_topic_for_publish(request_id, status):
     """
     :return: The topic for publishing method responses. It is of the format
     "$iothub/methods/res/<status>/?$rid=<requestId>
+
+    Note that all inputs MUST be strings (not ints!)
```

Comment:
>Note that all inputs MUST be strings (not ints!) [](start = 4, length = 48)

Why?  What breaks?

Reply:
`quote` needs a string or byte object, which an int is not.

I have now added forced string conversions to avoid this problem.

---
In reply to: [399660431](https://github.com/Azure/azure-iot-sdk-python/pull/514#discussion_r399660431) [](ancestors = 399660431)